lib/scoped/find.js

```diff
@@ -7,7 +7,9 @@ function find (type, api, objectsOrIds) {
 
   .then(function (storedObjects) {
     if (Array.isArray(storedObjects)) {
-      var typedObjects = storedObjects.filter(typeFilter.bind(null, type))
+      var typedObjects = storedObjects.map(function (storedObject) {
+        return typeFilter(type, storedObject) ? storedObject : new Error('Item not found')
```

Comment:
I wasn't sure if 'Item not found' was the correct error to throw here.  Would 'Item out of scope' or 'Incorrect item type' be better?